packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -15,146 +15,122 @@
-  github: GitHubAPI,
-  owner: string,
-  repo: string,
-  branch: string
-) {
-  try {
-    const data = (
-      await github.repos.getBranchProtection({
-        owner,
-        repo,
-        branch,
-      })
-    ).data;
-    return data;
-  } catch (err) {
-    return null;
-  }
-}
-
-export = (app: Application) => {
-  app.on(['pull_request.opened', 'pull_request.reopened'], async context => {
-    const config = (await context.config(
-      CONFIGURATION_FILE_PATH,
-      {}
-    )) as Configuration;
-
-    console.log(config);
-    const { owner, repo } = context.repo();
-
-    const branchProtection = await getBranchProtection(
-      context.github,
-      owner,
-      repo,
-      context.payload.pull_request.head.repo.default_branch
-    );
-
-    const configProtection = config.required_status_checks;
-
-    const commitParams = context.repo({
-      pull_number: context.payload.pull_request.number,
-      per_page: 100,
-    });
-
-    let commitsResponse: Response<PullsListCommitsResponseItem[]>;
-    try {
-      commitsResponse = await context.github.pulls.listCommits(commitParams);
-    } catch (err) {
-      console.info(err);
-      app.log.error(err);
-      return;
-    }
-
-    const commits = commitsResponse.data;
-
-    let checkParams = context.repo({
-      name: 'merge-on-green-readiness',
-      conclusion: 'success' as Conclusion,
-      head_sha: commits[commits.length - 1].sha,
-    });
-
-    if (!branchProtection) {
-      checkParams = context.repo({
-        head_sha: commits[commits.length - 1].sha,
-        name: 'merge-on-green-readiness',
-        conclusion: 'failure' as Conclusion,
-        output: {
-          title: 'You have no required status checks',
-          summary: 'Enforce branch protection on your repo.',
-          text:
-            'To add required status checks to your repository, please follow instructions in this link: \nhttps://help.github.com/en/github/administering-a-repository/enabling-required-status-checks\n' +
-            '\nIn order to add applications to your repository that will run check runs, please follow instructions here: \nhttps://developer.github.com/apps/installing-github-apps/\n' +
-            '\nLastly, please make sure that your required status checks are the same as the ones listed in your config file if you created one.',
-        },
-      });
+handler.listPRs = async function listPRs(): Promise<WatchPR[]> {
+  const query = datastore.createQuery(TABLE).order('created');
+  const [prs] = await datastore.runQuery(query);
+  const result: WatchPR[] = [];
+  for (const pr of prs) {
+    const created = new Date(pr.created).getTime();
+    const now = new Date().getTime();
+    console.info(`keystore data`, pr[datastore.KEY]);
+    const url = pr[datastore.KEY].name;
+    let state = 'continue';
+    if (now - created > MAX_TEST_TIME) {
+      console.warn(`deleting stale PR ${url}`);
+      await handler.removePR(url);
+      state = 'stop';
     }
```

Comment:
TODO: I'd prefer to not have a "list" method that has side effects - perhaps later refactor this to do the list, then have an explicit loop over the returned `WatchPR` objects that removes the expired ones.

We can probably clean this up in a follow-up PR.

Reply:
Added this as a to-do in the code!